pkg/deploy/apply.go

```diff
@@ -47,10 +47,43 @@ const (
 )
 
 var (
-	awaitCompletionAnnotation = resourceutil.GetMiaAnnotation("await-completion")
-	apply                     = withAwaitableResource(applyResource)
+	deleteBeforeApplyAnnotation = resourceutil.GetMiaAnnotation("delete-before-apply")
+	awaitCompletionAnnotation   = resourceutil.GetMiaAnnotation("await-completion")
+	apply                       = decorateApplyResource(withAwaitableResource, withDeletableResource)
 )
 
+func decorateApplyResource(decorators ...func(applyFunction) applyFunction) applyFunction {
+	res := applyResource
```

Comment:
We should rename it, because res seems to be a resource. It instead is a function to apply the resource

Reply:
:+1: